scipy/cluster/hierarchy.py

```diff
@@ -2107,29 +2107,29 @@ def is_valid_im(R, warning=False, throw=False, name=None):
     xp = array_namespace(R)
     R = _asarray(R, order='c', xp=xp)
     valid = True
-    name_str = "%r " % name if name else ''
+    name_str = f"{name!r} " if name else ''
     try:
         if R.dtype != xp.float64:
-            raise TypeError('Inconsistency matrix %smust contain doubles '
-                            '(double).' % name_str)
+            raise TypeError(f'Inconsistency matrix {name_str}must contain doubles '
+                            '(double).')
         if len(R.shape) != 2:
-            raise ValueError('Inconsistency matrix %smust have shape=2 (i.e. '
-                             'be two-dimensional).' % name_str)
+            raise ValueError(f'Inconsistency matrix {name_str} must have shape=2 (i.e. '
```

Comment:
Ah, missed that. However, a follow-up:
Personally, it looks nicer to me to have a space e.g. `{name_str} must` rather than `{name_str}must`. So if it's OK, I'd like to remove the space from the f-string in line 2110.

Reply:
@lucascolley, please let me know what you think, and I'll change accordingly.

Reply:
Didn't think about the `None` case. I'll follow the original one then. Thank you for the explanation!